### ISSUE TITLE
Expand user defined inputs in hyperparameter functions

### ISSUE THREAD

**Issue by ttseriotou** [ttseriotou/sig-networks#38]
- [x]  weight decay in Adam()
- [x] let the user define a fixed patience
- [x] let the user define batch size in the input
- [x] option to use TensorDataset() and DataLoader() pytorch classes.
- [x] let the user define the combination method and if they want to use conv1d or augmentation and their size.
- [x] Add argument in input to include or exclude time as part of the in signature path
